src/Catalyst.Core.Lib.Tests/UnitTests/P2P/PeerIdValidatorTests.cs

```diff
@@ -0,0 +1,124 @@
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Linq;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.P2P;
+using Catalyst.Core.Modules.Cryptography.BulletProofs;
+using Catalyst.Protocol.Peer;
+using Catalyst.TestUtils;
+using FluentAssertions;
+using Google.Protobuf;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Lib.Tests.UnitTests.P2P
+{
+    public sealed class PeerIdValidatorTests
+    {
+        private readonly ITestOutputHelper _output;
+        private readonly IPeerIdValidator _peerIdValidator;
+        private readonly PeerId _validPeerId;
+
+        public PeerIdValidatorTests(ITestOutputHelper output)
+        {
+            _output = output;
+            _validPeerId = PeerIdHelper.GetPeerId();
+            _peerIdValidator = new PeerIdValidator(new FfiWrapper());
+        }
+
+        [Fact]
+        public void Can_Validate_PeerId_Format()
+        {
+            _peerIdValidator.ValidatePeerIdFormat(_validPeerId);
+
+            _output.WriteLine(string.Join(" ", _validPeerId.ToByteArray()));
+            var fieldsInBytes = new[]
+            {
+                _validPeerId.Ip.ToByteArray(), BitConverter.GetBytes(_validPeerId.Port),
+                _validPeerId.PublicKey.ToByteArray()
+            };
+            _output.WriteLine(string.Join(" ", fieldsInBytes.SelectMany(b => b)));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(10)]
+        [InlineData(19)]
+        [InlineData(21)]
+        public void Can_Throw_Argument_Exception_On_Invalid_Public_Key(int pubKeySize)
+        {
+            var invalidPeer = new PeerId(_validPeerId)
+            {
+                PublicKey = new byte[pubKeySize].ToByteString()
+            };
+            new Action(() => _peerIdValidator.ValidatePeerIdFormat(invalidPeer))
+               .Should().Throw<ArgumentException>().WithMessage("*PublicKey*");
+
+            invalidPeer.PublicKey = new byte[21].ToByteString();
+        }
+
+        private sealed class IpTestData : TheoryData<byte[]>
+        {
+            public IpTestData()
+            {
+                Add(new byte[0]);
+                Add(new byte[4]);
+                Add(new byte[15]);
+                Add(new byte[17]);
+            }
+        }
+
+        [Theory]
+        [ClassData(typeof(IpTestData))]
+
+        //Todo: discuss if this is relevant: why do we enforce a given size for IPs (or anything) if proto handles it
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)